postgres/datadog_checks/postgres/postgres.py

```diff
@@ -157,8 +155,11 @@ def is_aurora(self):
 
     @property
     def resolved_hostname(self):
-        if self._resolved_hostname is None and self._config.dbm_enabled:
-            self._resolved_hostname = resolve_db_host(self._config.host)
+        if self._resolved_hostname is None:
```

Comment:
This needs to be

```python
if self._resolved_hostname is None and (self._config.dbm_enabled or self.config.disable_generic_tags):
    self._resolved_hostname = resolve_db_host(self._config.host)
```

If there's a single agent monitoring multiple remote hosts then by dropping the `server` tag we no longer have any way to differentiate them since they'll all be tagged with the same `agent_hostname`. 

This way when `disable_generic_tags: true` then the host tagging is also fixed to ensure all telemetry is tagged by the correct host and we don't need to rely on the server tag anymore.